packages/app/app/actions/downloads.ts

```diff
@@ -170,3 +170,34 @@ export const clearFinishedDownloads = createStandardAction(DownloadActionTypes.C
       payload: filteredTracks
     };
   });
+
+/**
+* Helper function to read the old track format into the new format.
+*
+* `Track.artist` and `Track.extraArtists` are written into {@link Track.artists}
+*/
+function getDownloadsBackwardsCompatible(): Download[] {
+  const downloads: Download[] = store.get('downloads');
+
+  downloads.forEach(download => {
+    // @ts-expect-error For backwards compatibility we're trying to parse an invalid field
+    if (download.track.artists || !download.track.artist) {
+      return;
+    }
+
+    // @ts-expect-error For backwards compatibility we're trying to parse an invalid field
+    if (download.track.artist) {
+      // @ts-expect-error For backwards compatibility we're trying to parse an invalid field
+      download.track.artists = _.isString(download.track.artist) ? [download.track.artist] : [download.track.artist.name];
+    }
+
+    // Assuming we have `extraArtists` on a track, we must had an `artist` which
+    // was already saved into `artists`, so this `track.artists` shouldn't be undefined
+    // @ts-expect-error For backwards compatibility we're trying to parse an invalid field
+    download.track.extraArtists?.forEach(artist => {
+      download.track.artists.push(artist);
+    });
```

Comment:
Instead of mutating by using `forEach`/`push`, you can use `map` and assign what `map` returns.

Reply:
This also applies to other places where `forEach` is used.

Reply:
Changed them all to use `map` now.
Aside from inside the helper functions there was only one single occurrence when reading from the DB.